block_code_demo.gd

```diff
@@ -4,10 +4,40 @@ extends Control
 
 @onready var _block_code := %BlockCode
 
+@onready var MainPanel := %MainPanel
 
 func _ready() -> void:
-	#var block_script: BlockScriptSerialization = 
-	#block_script.script_inherits = _get_custom_or_native_class(get_parent())
-	#block_script.generated_script = new_block_script.generated_script.replace("INHERIT_DEFAULT", new_block_script.script_inherits)
+	if _block_code.block_script == null:
+		var new_block_script: BlockScriptSerialization = load("res://addons/block_code/serialization/default_block_script.tres").duplicate(true)
+		new_block_script.script_inherits = _block_code._get_custom_or_native_class(_block_code.get_parent())
+		new_block_script.generated_script = new_block_script.generated_script.replace("INHERIT_DEFAULT", new_block_script.script_inherits)
+		_block_code.block_script = new_block_script
 
-	_context.set_block_code_node.call_deferred(_block_code)
+
+	#_context.set_block_code_node.call_deferred(_block_code)
+	await get_tree().process_frame
+	MainPanel.switch_block_code_node(_block_code)
+	MainPanel.script_window_requested.connect(script_window_requested)
+	print("_block_code ", _block_code)
+
+	var ShowScriptButton = MainPanel.get_node("%TitleBar").get_parent().get_node("ShowScriptButton")
+	ShowScriptButton.text = "Load Block Code"
+
+var showscriptwindowfirst = true
+const ScriptWindow := preload("res://addons/block_code/ui/script_window/script_window.tscn")
+func script_window_requested(scriptcontent):
+	if showscriptwindowfirst:
+		var script_window = ScriptWindow.instantiate()
+		script_window.script_content = scriptcontent
+		add_child(script_window)
+		await script_window.close_requested
+		script_window.queue_free()
+		script_window = null
+
+	var block_code_parent = _block_code.get_parent()
+	var script := GDScript.new()
+	script.set_source_code(scriptcontent)
+	script.reload()
+	block_code_parent.set_script(script)
+	block_code_parent._ready()
+	block_code_parent.set_process(true)
```

Comment:
Ooh, took me a minute to figure out what we're doing here, then I realized we handle `script_window_requested` in block_code_plugin.gd so this has to do it all over again. That seems like a bit of a mistake - we're happy spawning dialogs from main_panel.gd in other cases.

(Granted, using `EditorInterface.popup_dialog_centered()`, which is also going to be a mistake in this case).

As a note for later, might be good moving that whole concern into main_panel.gd, and then we should be able to drop all the stuff with script_window.tscn here.

I suspect _in the end_ we'll want to have a different main_panel.tscn altogether for this, which would probably _not_ have that node chooser option button at the top, and have a Play button instead. But to get there we'd need to break that scene up a bit as well (so we aren't duplicating a bunch of stuff), so I can appreciate that this is working with what we have at the moment. And, uh, indeed good to not refactor a bunch of things in this branch since it's rather behind main :)